packages/editor/src/store/effects/terms.js

```diff
@@ -0,0 +1,39 @@
+/**
+ * External dependencies
+ */
+import { get } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { dispatch, select } from '@wordpress/data';
+import { _x, sprintf, __ } from '@wordpress/i18n';
+import { speak } from '@wordpress/a11y';
+
+/**
+ * Internal dependencies
+ */
+import { resolveSelector } from './utils';
+
+/**
+ * Effect handler adding a new term to currently edited post.
+ *
+ * @param {Object} action  action object.
+ * @param {Object} store   Redux Store.
+ */
+export async function addTermToEditedPost( { slug, term } ) {
+	const taxonomy = await resolveSelector( 'core', 'getTaxonomy', slug );
+	const savedTerm = await dispatch( 'core' ).saveEntityRecord( 'taxonomy', slug, term );
```

Comment:
I guess in the end we don't escape the fact we use the promise return from the dispatch call.

Losing battle. Probably rather just have the `return` in `withDispatch` to avoid the inconsistency in what's returned from a dispatching action creator.

Reply:
We already return a promise from dispatch and this PR doesn't change it. I'm leaning towards avoid any changes to the data module API in this particular PR. We can tackle this issue in its dedicated PR